server/src/test/java/org/opensearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -345,7 +346,13 @@ private void testSimple(
         final int expectedCollectCount
     ) throws Exception {
         try (Directory directory = newDirectory()) {
-            try (RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory)) {
+            try (
+                RandomIndexWriter indexWriter = new RandomIndexWriter(
+                    random(),
+                    directory,
+                    newIndexWriterConfig().setMergePolicy(NoMergePolicy.INSTANCE)
```

Comment:
I have added test cases which can fail when the segment has deleted documents and the segment gets rid of them - 2 different code flows are triggered if there are/not deleted documents.  We did not had any test cases with deleted documents, neither did we had different code flows. So it makes sense that we add no merge policy on the index so that the test-index does not changes it s state in between.

And the change https://github.com/opensearch-project/OpenSearch/pull/11643 was merged some 18 hours ago, so I guess we caught this test case pretty early.

Reply:
> We did not had any test cases with deleted documents, neither did we had different code flows. So it makes sense that we add no merge policy on the index so that the test-index does not changes it s state in between.

Thank you, could we extract them (new test cases) into separate test suite?

Reply:
I don't think it make sense to segregate the test cases based on just internal optimizations inclusions. Currently, the tests are included in TermsAggregatorTests and have been segregated into different test methods. The new test cases have been added to make sure the optimized workflow in case of no deleted documents in the segment in utilized. And also, the old unoptimized workflow in case of deleted documents. Earlier, there were no deleted documents in tests and there was a single aggregation logic.

Reply:
There is apparently an issue with new test cases & merge policy, which was (and per my understanding, still is) non issue before, why we disable the merge for all tests? We thrive to keep tests as close to real world usage as possible.

Reply:
We are **not disabling** the merge policy for all cases, we are only disabling in this test method(s) so as to segregate the 2 cases. It won't be possible to gracefully add coverage for both cases otherwise.

Reply:
Oops sorry about that, I should have looked that it is test specific (Github shows only small diff), thanks @sandeshkr419 !